From 63433db3b06540fb45917ae210e08d6389d16054 Mon Sep 17 00:00:00 2001
From: Jonathan Marek <jonathan@marek.ca>
Date: Sat, 5 Oct 2019 21:15:07 -0400
Subject: [PATCH 09/20] HACK: delay rpm init

Delay RPM init. This is a hack that somehow fixes some problems.
Hopefully the need for this patches goes away once the IOMMU is
working.

Signed-off-by: Jonathan Marek <jonathan@marek.ca>
Signed-off-by: Brian Masney <masneyb@onstation.org>
---
 drivers/clk/qcom/clk-smd-rpm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/clk/qcom/clk-smd-rpm.c b/drivers/clk/qcom/clk-smd-rpm.c
index 0bbfef9fa6de..447ca887deb1 100644
--- a/drivers/clk/qcom/clk-smd-rpm.c
+++ b/drivers/clk/qcom/clk-smd-rpm.c
@@ -15,6 +15,7 @@
 #include <linux/of_device.h>
 #include <linux/platform_device.h>
 #include <linux/soc/qcom/smd-rpm.h>
+#include <linux/delay.h>
 
 #include <dt-bindings/clock/qcom,rpmcc.h>
 #include <dt-bindings/mfd/qcom-rpm.h>
@@ -770,6 +771,9 @@ static int rpm_smd_clk_probe(struct platform_device *pdev)
 	rcc->clks = rpm_smd_clks;
 	rcc->num_clks = num_clks;
 
+	/* delay rpm init, somehow this fixes some problems */
+	usleep_range(100000, 110000);
+
 	for (i = 0; i < num_clks; i++) {
 		if (!rpm_smd_clks[i])
 			continue;
-- 
2.25.1

